docs/getting-started/quick-start.md

```diff
@@ -250,6 +280,51 @@ fun main() {
 }
 ```
 
+### IOS with SwiftUI
+
+1. Create `RootHolder` class that holds the root component and its lifecycle.
+
+```swift
+class RootHolder : ObservableObject {
+    let lifecycle: LifecycleRegistry
+    let root: RootComponent
+
+    init() {
+        lifecycle = LifecycleRegistryKt.LifecycleRegistry()
+
+        root = DefaultRootComponent(
+            componentContext: DefaultComponentContext(
+                lifecycle: lifecycle,
+            ),
+        )
+
+        lifecycle.onCreate()
+    }
+
+    deinit {
+        // Destroy the root component before it is deallocated
+        lifecycle.onDestroy()
+    }
+}
+```
+
+2. Create `RootHolder` instance and pass it to the `RootView` constructor.
+
+```swift
+@main
+struct app_iosApp: App {
+    private var rootHolder: RootHolder { RootHolder() }
+        
+    var body: some Scene {
+        WindowGroup {
+            RootView(rootHolder.root)
+                .onAppear { LifecycleRegistryExtKt.resume(self.rootHolder.lifecycle) }
+                .onDisappear { LifecycleRegistryExtKt.stop(self.rootHolder.lifecycle) }
+        }
```

Comment:
```suggestion
    private var rootHolder: RootHolder { RootHolder() }
    @Environment(\.scenePhase) var scenePhase
        
    var body: some Scene {
        WindowGroup {
            RootView(rootHolder.root)
                .onAppear { LifecycleRegistryExtKt.start(self.rootHolder.lifecycle) }
                .onDisappear { LifecycleRegistryExtKt.stop(self.rootHolder.lifecycle) }
                .onChange(of: scenePhase) { newPhase in
                    switch newPhase {
                    case .background:
                        LifecycleRegistryExtKt.stop(rootHolder.lifecycle)
                    case .active:
                        LifecycleRegistryExtKt.start(rootHolder.lifecycle)
                    case .inactive:
                        LifecycleRegistryExtKt.pause(rootHolder.lifecycle)
                    @unknown default: break
                    }
                }
        }
```

Sorry for interrupting and correct me if I miss something @arkivanov, but your `sample` app and the docs states resuming (instead of starting) on `onAppear` and stopping (which sounds right) on `onDissapear`.  I had some issues when I put my app in the background and after that when I open the app again looks like it skips the `start` and goes to `resume` instead (I'm not sure about this in particular just following the logic). I'm coming from Android and I suggest a bit more precise control over the lifecycle.

Let me know if I miss something and my solution is breaking something else or is not needed overall. If my approach looks more robust I can fix the sample app quickly via another PR.

Reply:
No problem at all! Asking questions is always good!

Calling `resume` should automatically drive the lifecycle through `start`, if needed. So initially `onAppear` calls `resume` (which calls `onCreate` -> `onStart` -> `onResume`), then `onDisappear` calls `stop` (which calls `onPause` -> `onStop`), then `onAppear` calls `resume` again (which calls `onStart` -> `onResume`). See the source code [here](https://github.com/arkivanov/Essenty/blob/master/lifecycle/src/commonMain/kotlin/com/arkivanov/essenty/lifecycle/LifecycleRegistryExt.kt#L29).

On the other hand in your proposed change, you never call `resume`, which means `onResume` will never be called.

Reply:
Ah, got it! So it's sufficient on `.background` to call `stop` and on `.active` to call `resume`. I want to emphases on `scenePhase` as `onAppear` and `onDisappear` behaves like more like onCreate and onDestroy on Android.

Reply:
Yes, it should be fine. Btw, what about onDisapper? Isn't it gets called when the app goes to background?

Reply:
No, only when you swipe and "kill" the app :(

Reply:
Oh, this is something I should fix!